### ISSUE TITLE
Argument: Rust is no better than mastodon?

### ISSUE THREAD

**Issue by SoniEx2** [rustodon/rustodon#58]
https://github.com/rust-lang/rfcs/pull/1925

It's possible to take the package in a different direction from the ecosystem, but it's not gonna be easy. Rust itself doesn't really accept user feedback, which can cause some real issues sometimes.

I like to believe it's gonna be different this time, and I hope for the best, but I'm still a bit worried about it.

**Comment by barzamin** [comment]
1. I am unsure how the linked RFC relates to this project.
2. Rust definitely accepts user feedback - see the evolution since its inception.
3. Making a copy of Mastodon is not our goal - the name Rustodon was chosen as an interim term, and should not imply that we are attempting to recreate it.

In short, I do not believe the problems you describe with the Rust ecosystem exist to the stated degree, and, additionally, I don't think they are _relevant to our project_.

As a result, I am closing this issue.

**Comment by yipdw** [comment]
That thread you linked to is about an optional language extension to Rust, and I don't think it has any impact on Rustodon proper.  (Rustodon code in no way has to use that extension, and it probably won't ever -- that extension has been in Rust stable for a while now and I can't think of an instance where match bars before or after has been an issue for us.)

**Comment by barzamin** [comment]
@SoniEx2 I have had more time to think about this overnight, and I'm going to explain why we've locked this in more detail now.

I believe the argument you're making is that building software on Rust might be a mistake, because you perceive certain issues with the community and ecosystem.

I would strongly argue that, now that Rust is somewhat stabilized, choosing it over any other sufficiently advanced language is simply a matter of taste, and criticizing our choice of language based on community issues _with the language_ is out of scope for both our issue tracker _and_ the project itself.

---

Even though it isn't strictly necessary, I feel some reason to justify our choice of language.
 
I picked Rust when I started working on this because it's a cool language that I like. I don't think the community is failing and I think the ecosystem - see, Rustdoc, Crates.io, &c - is _fantastic_. I agree there are issues with the RFC process - see the Unicode Identifiers proposal and the discussion therein - but the language designers seem to be careful, and I have strong hopes that Rust's growth can strike a balance between stability and feature-richness.

---

In short, using Rust has not limited our ability to build a good system.
